vlo-web-app/src/main/java/eu/clarin/cmdi/vlo/wicket/provider/SolrDocumentExpansionPairProvider.java

```diff
@@ -22,36 +22,84 @@
+import org.slf4j.LoggerFactory;
+
+import com.sun.istack.logging.Logger;
+
+import org.apache.wicket.protocol.http.servlet.ServletWebRequest;
+import eu.clarin.cmdi.vlo.pojo.QueryFacetsSelection;
+import eu.clarin.cmdi.vlo.exposure.models.SearchQuery;
+import eu.clarin.cmdi.vlo.exposure.models.SearchResult;
+import eu.clarin.cmdi.vlo.config.ServletVloConfigFactory;
+import eu.clarin.cmdi.vlo.config.VloConfig;
+
 /**
  *
  * @author twagoo
  */
 public class SolrDocumentExpansionPairProvider implements IDataProvider<SolrDocumentExpansionPair> {
+    private final static org.slf4j.Logger logger = LoggerFactory.getLogger(SolrDocumentExpansionPairProvider.class);
 
     private final FieldNameService fieldNameService;
 
     private final IModel<QueryFacetsSelection> selectionModel;
 
     private Long size;
 
-    public SolrDocumentExpansionPairProvider(IModel<QueryFacetsSelection> selection, FieldNameService fieldNameService) {
+    public SolrDocumentExpansionPairProvider(IModel<QueryFacetsSelection> selection,
+            FieldNameService fieldNameService) {
         this.selectionModel = selection;
         this.fieldNameService = fieldNameService;
     }
 
     @Override
     public Iterator<? extends SolrDocumentExpansionPair> iterator(long first, long count) {
-        final SolrDocumentExpansionList documents = getDocumentService().getDocumentsWithExpansion(selectionModel.getObject(),
-                BigDecimal.valueOf(first).intValueExact(), // safe long->int conversion
-                BigDecimal.valueOf(count).intValueExact(),
-                FacetConstants.COLLAPSE_FIELD_NAME); // safe long->int conversion
+        final SolrDocumentExpansionList documents = getDocumentService().getDocumentsWithExpansion(
+                selectionModel.getObject(), BigDecimal.valueOf(first).intValueExact(), // safe long->int conversion
+                BigDecimal.valueOf(count).intValueExact(), FacetConstants.COLLAPSE_FIELD_NAME); // safe long->int
+                                                                                                // conversion
+
+        VloConfig vloConfig = VloWicketApplication.get().getVloConfig();
+        if (vloConfig.isVloExposureEnabled()) {
+            try {
+                // these values to be saved in postgresql to calculate record-exposure
+                // get page url
+                String pageUrl = ((ServletWebRequest) RequestCycle.get().getRequest()).getContainerRequest()
+                        .getRequestURL().toString();
+                // get user ip address
+                String ip = ((WebClientInfo) VloWebSession.get().getClientInfo()).getProperties().getRemoteAddress();
+                QueryFacetsSelection selection = this.selectionModel.getObject();
+                // get search term
+                String searchTerm = this.selectionModel.getObject().getQuery();
+                List<SearchResult> res = new ArrayList<SearchResult>();
```

Comment:
could be that performance benefits from initialising with the size of the array, which is known